ferrocene/doc/requirements/src/self-test.rst

```diff
@@ -0,0 +1,180 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+``ferrocene-self-test`` tool
+============================
+
+General checks
+--------------
+
+Checks for executables
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+   **executable** refers to the following:
+
+   - ``rustc``
+   - ``rustdoc``
+
+The executable should exist in the ``bin`` directory, relative to the root of the installation.
+
+The executable should be a regular file.
+
+The executable should have read and execute permissions set for all users.
+
+When executed with ``-vV`` command-line options, the following apply:
+
+- The executable should exit successfully.
+
+- The output of the executable should be UTF-8 text.
+
+- The output of the executable should have at least 3 key-value pairs,
+  with these 3 keys: ``host``, ``commit-hash``, and ``release``.
+
+  Example output for rustc:
+
+  .. code-block:: text
+
+     commit-hash: f228f60438264270168a59eba2bc4aaca0ca7c02
+     host: x86_64-unknown-linux-gnu
+     release: 1.81.0-nightly
+
+- The value of ``host`` from the output of the executable should match the target-triple
+  of the executing environment.
+
+- The value of ``commit-hash`` from the output of the executable should match the 
+  commit hash of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+- The value of ``release`` from the output of the executable should match the
+  version of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+Checks for targets
+------------------
+
+
+.. note::
+
+   In the subsections below, we have these substitutions:
+
+   - ``$target`` refers to any of the targets listed
+   - ``$hash`` is a 16 character hexadecimal string
+
+All targets
+^^^^^^^^^^^
+
+.. note::
+
+   These checks are for all locally-installed targets listed in :doc:`user-manual:targets/index`.
+
+Inside of ``lib/rustlib/$target/lib`` directory,
+relative to the root of the installation,
+should exist these regular files:
+
+- ``libcore-$hash.rlib``
+- ``liballoc-$hash.rlib``
+
+Inside of ``lib/rustlib/$target/lib`` directory,
+relative to the root of the installation,
+should not exist duplicates of these regular files:
+
+- ``libcore-$hash.rlib``
+- ``liballoc-$hash.rlib``
```

Comment:
Maybe we can simplify it like this?
```suggestion
No duplicates of these files should exist.
```

Reply:
Maybe there is a different approach to avoid duplication.

Reply:
ccc7515e918dcd9cd88f58ce969aa0a9a125dacc